src/node_api.cc

```diff
@@ -3571,6 +3571,91 @@ napi_status napi_get_arraybuffer_info(napi_env env,
   return napi_clear_last_error(env);
 }
 
+napi_status napi_is_sharedarraybuffer(napi_env env, napi_value value, bool* result) {
+  CHECK_ENV(env);
+  CHECK_ARG(env, value);
+  CHECK_ARG(env, result);
+
+  v8::Local<v8::Value> val = v8impl::V8LocalValueFromJsValue(value);
+  *result = val->IsSharedArrayBuffer();
+
+  return napi_clear_last_error(env);
+}
+
+napi_status napi_create_sharedarraybuffer(napi_env env,
+                                    size_t byte_length,
+                                    void** data,
+                                    napi_value* result) {
+  NAPI_PREAMBLE(env);
+  CHECK_ARG(env, result);
+
+  v8::Isolate* isolate = env->isolate;
+  v8::Local<v8::SharedArrayBuffer> buffer =
+      v8::SharedArrayBuffer::New(isolate, byte_length);
+
+  // Optionally return a pointer to the buffer's data, to avoid another call to
+  // retrieve it.
+  if (data != nullptr) {
+    *data = buffer->GetContents().Data();
+  }
+
+  *result = v8impl::JsValueFromV8LocalValue(buffer);
+  return GET_RETURN_STATUS(env);
+}
+
+napi_status napi_create_external_sharedarraybuffer(napi_env env,
```

Comment:
please remove this one as it isn't something that we can expect other engines to be able to implement

Reply:
@devsnek Just for context, we also have `napi_create_external_arraybuffer`

Reply:
@addaleax then let's not make the same mistake twice?

Reply:
So we have to wait until any other engines stably release the SharedArrayBuffer feature? Edge, FF, Safari can use it with experimental flag by the way.

Reply:
So the other engines do not support **external** arraybuffer and n-api has already some mis-designe related to external-arraybuffer api things?